kie-maven-plugin/src/main/java/org/kie/maven/plugin/BuildMojo.java

```diff
@@ -158,19 +208,53 @@ private void buildDrl() throws MojoFailureException {
 
         try {
             setSystemProperties(properties);
-            KieRepository kr = ks.getRepository();
-            InternalKieModule kModule = (InternalKieModule) kr.addKieModule(ks.getResources().newFileSystemResource(sourceFolder));
-            for (InternalKieModule kmoduleDep : kmoduleDeps) {
-                kModule.addKieDependency(kmoduleDep);
-            }
+            List<File> resources = getResourceFiles(sourceFolder);
+            ResultsImpl messages = null;
+            InternalKieModule kModule = null;
+            KieProject kieProject = null;
+            if (containsPMML) {
+                KieFileSystem kfs = ks.newKieFileSystem();
+                for (File file: resources) {
+                    if (!file.getPath().contains("META-INF")) {
+                        Resource fresource = ResourceFactory.newFileResource(file);
+                        kfs.write(fresource);
+                    }
+                }
+                final KieBuilderImpl kieBuilder = new KieBuilderImpl(kfs);
+                kieBuilder.buildAll();
+                messages = (ResultsImpl)kieBuilder.getResults();
+                kModule = (InternalKieModule)kieBuilder.getKieModule();
+                List<String> classFiles = kModule.getFileNames()
+                        .stream()
+                        .filter(name -> name.endsWith(".class"))
+                        .collect(Collectors.toList());
+
+                MemoryFileSystem mfs = kModule instanceof CanonicalKieModule ?
```

Comment:
@lanceleverich what in my opinion is not entirely correct is that you pasted parts of the executable model mojo here. I don't think this makes sense even because in case of a generated model this mojo will be completely skipped (see line 110). I'm still hoping that we could find a way to unify the build with and without pmml. Can we please have a call this afternoon (your morning) to discuss this?